### PR TITLE
`BlockId` removal: `Client::runtime_version_at`

### PR DESCRIPTION
**Pull request by michalkucharczyk**
This PR changes the arguments of `Client::runtime_version_at` from: `BlockId<Block>` to: `Block::Hash`

This PR is part of BlockId::Number refactoring analysis (https://github.com/paritytech/substrate/issues/11292)

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

File: bin/node/cli/src/service.rs [hunk 1]
```diff
@@ -754,9 +754,9 @@ mod tests {
 				let to: Address = AccountPublic::from(bob.public()).into_account().into();
 				let from: Address = AccountPublic::from(charlie.public()).into_account().into();
 				let genesis_hash = service.client().block_hash(0).unwrap().unwrap();
-				let best_block_id = BlockId::number(service.client().chain_info().best_number);
+				let best_block_hash = service.client().chain_info().best_hash;
```
Comment (melekes) [inline bin/node/cli/src/service.rs:757]:
```suggestion
				let best_hash = service.client().chain_info().best_hash;
```

File: bin/node/cli/src/service.rs [hunk 2]
```diff
@@ -754,9 +754,9 @@ mod tests {
...
 				let (spec_version, transaction_version) = {
-					let version = service.client().runtime_version_at(&best_block_id).unwrap();
+					let version = service.client().runtime_version_at(best_block_hash).unwrap();
```
Comment (melekes) [inline bin/node/cli/src/service.rs:759]:
```suggestion
					let version = service.client().runtime_version_at(best_hash).unwrap();
```

**Review by melekes — APPROVED** [review]
👍

**Comment by michalkucharczyk** [comment]
bot rebase

**Comment by michalkucharczyk** [comment]
bot merge

**Pull request merged**